src/service.ts

```diff
@@ -206,14 +206,15 @@ export const snowballRService: ISnowballR = {
         const { oldPassword, newPassword } = call.request;
         const user = getAuthenticated(call.metadata)!;
 
-        if (user.password == oldPassword) {
-            USERS.get(user.email)!.password = newPassword;
-            callback(null, {});
-        } else {
+        if (user.password != oldPassword) {
             callback({
                 code: status.UNAUTHENTICATED,
                 details: "The old password is incorrect",
             });
+            return;
+        } else {
+            USERS.get(user.email)!.password = newPassword;
+            callback(null, {});
         }
```

Comment:
This `return` is redundant if you use the `else` block. Here and the other changes you made.
What about a correct guard clause?
```suggestion
            return;
        }

        USERS.get(user.email)!.password = newPassword;
        callback(null, {});
```

Reply:
I can correct this.